hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/DeleteMarkerProcedure.scala

```diff
@@ -60,6 +62,9 @@ class DeleteMarkerProcedure extends BaseProcedure with ProcedureBuilder with Log
         true
       case Failure(e) =>
```

Comment:
if success the client will not be closed?

Reply:
fixed and closed the write client in both success and exception scenarios